src/main/java/com/google/gcp_variant_transforms/library/SchemaGeneratorImpl.java

```diff
@@ -17,65 +22,148 @@
+        .setMode(SchemaUtils.constantFieldNameToModeMap.get(fieldName))
+        .setType(SchemaUtils.constantFieldNameToTypeMap.get(fieldName))
+        .setFields(subFields);
+  }
+
+  /**
+   * Creates and returns a Field from a compound header line
+   * @param headerLine
+   * @return an INFO or FORMAT Field
+   */
+  @VisibleForTesting
+  protected TableFieldSchema convertCompoundHeaderLineToField(VCFCompoundHeaderLine headerLine) {
+    String bqFieldMode = SchemaUtils.BQFieldMode.REPEATED; // Number = A, R, G, or >1
+    if (headerLine.getCountType() == VCFHeaderLineCount.INTEGER &&
+        headerLine.getCount() <= 1){
+      bqFieldMode = SchemaUtils.BQFieldMode.NULLABLE;
+    }
+    return new TableFieldSchema()
+        .setName(SchemaUtils.getSanitizedFieldName(headerLine.getID()))
+        .setDescription(headerLine.getDescription())
+        .setMode(bqFieldMode)
+        .setType(SchemaUtils.HTSJDKTypeToBQTypeMap.get(headerLine.getType()));
   }
 
   /**
-   * Creates and returns Call record's subfields.
-   * @return ImmutableList<TableFieldSchema>
+   * Creates and returns a Call record's subfields.
+   * @param vcfHeader
+   * @return a list of call sub-fields
    */
   @VisibleForTesting
-  protected ImmutableList<TableFieldSchema> getCallSubFields() {
-    ImmutableList.Builder<TableFieldSchema> callSubFieldsBuilder = new ImmutableList.Builder<TableFieldSchema>();
-    Collection<String> fieldNames = SchemaUtils.callsSubFieldIndexToNameMap.values();
-    for (String fieldName : fieldNames) {
+  protected ImmutableList<TableFieldSchema> getCallSubFields(VCFHeader vcfHeader) {
+    ImmutableList.Builder<TableFieldSchema> callSubFields =
+        new ImmutableList.Builder<>();
+    Collection<String> callFieldNames = SchemaUtils.callsSubFieldIndexToNameMap.values();
+    for (String fieldName : callFieldNames) {
       TableFieldSchema callSubField = new TableFieldSchema()
-            .setName(fieldName)
-            .setMode(SchemaUtils.callSubFieldNameToModeMap.get(fieldName))
-            .setDescription(SchemaUtils.callSubFieldNameToDescriptionMap.get(fieldName))
-            .setType(SchemaUtils.callSubFieldNameToTypeMap.get(fieldName));
-      callSubFieldsBuilder.add(callSubField);
+          .setName(fieldName)
+          .setDescription(SchemaUtils.callSubFieldNameToDescriptionMap.get(fieldName))
+          .setMode(SchemaUtils.callSubFieldNameToModeMap.get(fieldName))
+          .setType(SchemaUtils.callSubFieldNameToTypeMap.get(fieldName));
+      callSubFields.add(callSubField);
+    }
+    // Adds the FORMAT fields under the Calls record
+    for (VCFFormatHeaderLine formatHeaderLine : vcfHeader.getFormatHeaderLines()){
+        // Phaseset and Genotype are already added to callSubFields
+        String callSubFieldName = formatHeaderLine.getID();
+        if (!callSubFieldName.equals(VCFConstants.GENOTYPE_KEY)
+            && !callSubFieldName.equals(VCFConstants.PHASE_SET_KEY)) {
```

Comment:
@BoscoSuen Please note in the schema instead of `VCFConstants.GENOTYPE_KEY` we use `CALLS_GENOTYPE` as column name.
Similarly we use `CALLS_PHASESET` instead of `VCFConstants.PHASE_SET_KEY`.

We should take this into account when we create the BQRows.

Reply:
Yeah. `VCFConstants.PHASE_SET_KEY` is "PS" in the VCF record field and `CALLS_PHASESET` is "phaseset" in BQ schema field name. Correct me if I am wrong.